File: Data Structures & Algorithms/max-consecutive-ones/submission-1.py
class Solution:
    def findMaxConsecutiveOnes(self, nums: List[int]) -> int:
        
        """
        time: O(n)
        space: O(1)
        """
        res = cnt = 0
        for num in nums:
            cnt += 1 if num else -cnt
            res = max(res, cnt)
        return res

File: Data Structures & Algorithms/duplicate-integer/submission-1.java
class Solution {
    public boolean hasDuplicate(int[] nums) {
        Map<Integer, Integer> numMap = new HashMap<>();

        /**
        Time: O(n)
        Space: O(n)
        */
        for (int num: nums) {
            if (numMap.containsKey(num)) {
                return true;
            }
            numMap.put(num, 1);
        }

        return false;
    }
}